### PR TITLE
Feature/14059 spa feedback

### PR DESCRIPTION
**Pull request by pris54**
Feedback form, please see the story for further details.

### CODE REVIEW

File: pupil-spa/src/app/feedback/feedback.component.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+  private pupilData: object;
+  private feedbackExists: boolean;
+  private feedbackData: object;
+  private submitted: boolean;
+
+  constructor(private router: Router, private storageService: StorageService) {}
+
+  ngOnInit() {
+    if (!this.validate()) {
```
Comment (jon-shipley) [inline pupil-spa/src/app/feedback/feedback.component.ts]: I find `validate()` confusing in this context (form validation / submission) as really this is checking whether feedback has already been given.  How about calling it `componentValidate()` instead?

File: pupil-spa/src/app/feedback/feedback.component.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    if (this.selectedInputType !== undefined && this.selectedSatisfactionRating !== undefined) {
+      this.enableSubmit = true;
+    }
+  }
+
+  onSubmit(commentText) {
+    if (this.submitted === true) {
+      return;
+    }
+    this.submitted = false;
```
Comment (jon-shipley) [inline pupil-spa/src/app/feedback/feedback.component.ts]: Shouldn't this be setting `this.submitted = true`?  Is there a unit test for duplicate submission?

Comment (GeorgeChatzigiannis) [reply pupil-spa/src/app/feedback/feedback.component.ts]: I don't see any test around this myself.
@pris54 do you think it should be covered?

Comment (pris54) [reply pupil-spa/src/app/feedback/feedback.component.ts]: That's correct. ```this.submitted``` defaults to false and remains false until a successful submission happens (line 96).

File: pupil-spa/src/app/feedback/feedback.component.ts (hunk 1 above)
Comment (pris54) [reply pupil-spa/src/app/feedback/feedback.component.ts]: OK, I've renamed it as suggested to ```componentValidate```

File: pupil-spa/src/app/feedback/feedback.component.ts (hunk 2 above)
Comment (jon-shipley) [reply pupil-spa/src/app/feedback/feedback.component.ts]: It was already set to false in `ngOnInit()` - why does it need to be set again?

Comment (pris54) [reply pupil-spa/src/app/feedback/feedback.component.ts]: @jon-shipley True, line removed.

File: pupil-spa/src/app/phase-banner/phase-banner.component.html [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <div class="phase-banner">
   <p>
     <strong class="phase-tag">BETA</strong>
-    <span>This is a new service – your <a href="#">feedback</a> will help us to improve it.</span>
+    <span>This is a new service – your <a href="/feedback">feedback</a> will help us to improve it.</span>
```
Comment (jon-shipley) [inline pupil-spa/src/app/phase-banner/phase-banner.component.html]: This link doesn't work - should be routerLink

File: pupil-spa/src/app/feedback-thanks/feedback-thanks.component.html [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<div class="grid-row">
+  <div class="column-two-thirds">
+    <header class="page-header">
+      <h1 class="heading-xlarge">Thank you for your feedback</h1>
+    </header>
+
+    <p class="form-buttons">
+      <a id='sign-out' href="/sign-out" class="button">Next pupil</a>
```
Comment (jon-shipley) [inline pupil-spa/src/app/feedback-thanks/feedback-thanks.component.html]: href should be routerLink

**Pull request merged**